### PR TITLE
Milestone 2: Model

### PR DESCRIPTION
**Pull request by DuaneDave**
### In this milestone:

- Data were processed in the model for the users DB
- Data were processed in the model for the likes DB
- Data were processed in the model for the posts DB
- Data were processed in the model for the comments DB

#### Custom methods were added also for each classes

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Post < ActiveRecord::Base
+  belongs_to :user
+  has_many :comments
+  has_many :likes
+
+  def update_posts_counter
+    user.increment!(:post_counter)
+  end
```
Comment (EricMbouwe) [inline app/models/post.rb:8]: - [OPTIONAL] You can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_post_counter` method every time a new post has been saved to database, then make the `update_post_counter` method privateStudents can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_post_counter method every time a new post has been saved to database, then make the update_post_counter method private

File: app/models/like.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class Like < ActiveRecord::Base
+  belongs_to :user
+  belongs_to :post
+
+  def update_likes_counter
+    post.increment!(:likes_count)
+  end
```
Comment (EricMbouwe) [inline app/models/like.rb:7]: - [OPTIONAL] As mentioned above you can use `after_save :update_likes_counter`(https://github.com/microverse/guides/tse/code_review/code_review_guidelines_2_0/-/issues/4), after_save :update_comments_counter

File: app/models/comment.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class Comment < ActiveRecord::Base
+  belongs_to :user
+  belongs_to :post
+
+  def update_comments_counter
+    post.increment!(:comments_count)
+  end
```
Comment (EricMbouwe) [inline app/models/comment.rb:7]: - [OPTIONAL] As mentioned above you can use `after_save :update_comments_counter`(https://github.com/microverse/guides/tse/code_review/code_review_guidelines_2_0/-/issues/4), after_save :update_comments_counter

**Review by EricMbouwe — APPROVED** [review]
Hi @DuaneDave, 

Your project is complete! ⚡🔥  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 🎉🔥
- Model associations ✔
- Model methods ✔
- No linter errors ✔

## Optional suggestions ☂

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/comment.rb (hunk 3 above)
Comment (DuaneDave) [reply app/models/comment.rb:7]: thank you @EricMbouwe for the suggestions, I will look into the article

**Pull request merged**